cfg_mgmt/github.py

```diff
@@ -0,0 +1,326 @@
+
+
+def _push_to_github(
+    cfg_dir: str,
+    config_element,
+    git_helper,
+    target_ref,
+):
+    repo = git_helper.repo
+
+    actor = git.Actor(
+        name=git_helper.github_cfg.credentials().username(),
+        email=git_helper.github_cfg.credentials().email_address(),
+    )
+
+    repo.git.add(os.path.abspath(cfg_dir))
+    repo.index.commit(
+        _pretty_commit_message(config_element),
+        author=actor,
+        committer=actor,
+    )
+
+    try:
+        logger.info('Pushing changed config ...')
+        git_helper.push(
+            from_ref='@',
+            to_ref=target_ref,
+        )
+    except:
+        logger.error('Failed to push.')
+        # undo local changes
+        git_helper.repo.git.reset('--hard', '@~')
+        raise
+
+
+def _update_config_status(
+    cfg_dir: str,
+    cfg_status_filename: str,
+    config_element: model.NamedModelElement,
+    config_statuses: typing.Iterable[CfgStatus],
+):
+    for cfg_status in config_statuses:
+        if cfg_status.matches(
+            element=config_element,
+        ):
+            break
+    else:
+        # does not exist
+        cfg_status = CfgStatus(
+            target=CfgTarget(
+                type=config_element._type_name,
+                name=config_element.name(),
+            ),
+            credential_update_timestamp=datetime.date.today().isoformat(),
+        )
+        config_statuses.append(cfg_status)
+    cfg_status.credential_update_timestamp = datetime.date.today().isoformat()
+
+    with open(os.path.join(cfg_dir, cfg_status_filename), 'w') as f:
+        yaml.dump(
+            {
+                'config_status': [
+                    dataclasses.asdict(cfg_status)
+                    for cfg_status in config_statuses
+                ]
+            },
+            f,
+        )
+
+
+def _update_config_queue(
+    cfg_dir: str,
+    queue_file_name: str,
+    config_element: model.NamedModelElement,
+    config_queue: dict,
+    config_name: str,
+    secret_id: typing.Dict[str, typing.List[dict]],
+):
+    entry = CfgQueueEntry(
+        target=CfgTarget(
+            name=config_name,
+            type=config_element._type_name,
+        ),
+        deleteAfter=(datetime.datetime.today() + datetime.timedelta(days=7)).date().isoformat(),
+        secretId=secret_id,
+    )
+    config_queue['rotation-queue'].append(
+        dataclasses.asdict(entry),
+    )
+    _write_config_queue(cfg_dir=cfg_dir, queue_file_name=queue_file_name, config_queue=config_queue)
+
+
+def update_user(
+    github_config: GithubConfig,
+    username: str,
+) -> str:
+    credential = github_config.credentials(technical_user_name=username)
+    gh_api = ccc.github.github_api(github_config, username=username)
+    old_private_key = credential.private_key()
+
+    # current private key uses EdDSA which is not supported by Pycryptodome
+    try:
+        old_public_key = _corresponding_public_key(old_private_key)
+    except ValueError:
```

Comment:
is there a more specific execption available? also, I would at least log a warning in this case. also: let us add a note that we plan to remove this try-execpt after first rotation

Reply:
Added a TODO, but unfortunately that is the error raised by Pycryptodome.